### ISSUE TITLE
Build failure: gettext

### ISSUE THREAD

**Issue by axelkar** [NixOS/nixpkgs#269172]
### Steps To Reproduce

1. Import nixpkgs with `crossSystem = lib.systems.examples.aarch64-android-prebuilt;`
2. Use `gettext` or any package which depends on it, like `gcc`, `binutils` or `binutils-unwrapped`.

In short: `nix build github:axelkar/min-gettext-bug-producer`

### Build log

https://gist.github.com/axelkar/49e60e508ef5dc8115091080575a350b

important:
```log
./dcigettext.c:147:7: error: redeclaration of 'getcwd' must have the 'overloadable' attribute
char *getcwd ();
      ^
/nix/store/yjig1z72cqigrxlykkvm7i3235d7y3ag-aarch64-unknown-linux-android-ndk-toolchain-24.0.8215888/bin/../sysroot/usr/include/bits/fortify/unistd.h:105:7: note: previous overload of function is here
char* getcwd(char* const __pass_object_size buf, size_t size)
      ^
1 error generated.
make[5]: *** [Makefile:2144: dcigettext.lo] Error 1
```

### Notify maintainers

@zimbatm 
@vrthra 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.5.4, NixOS, 23.11 (Tapir), 23.11.20230919.5ba549e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.17.0`
 - channels(root): `"nixos-23.05"`
 - nixpkgs: `/nix/store/cz973qsjldbw4x7fx0rwcvhr9k645vz2-source`
```

### Priorities

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

btw, is there a prebuilt gcc with host = `x86_64-linux` and target = `aarch64-android-prebuilt` in the NixOS Hydra or somewhere else?

**Comment by zimbatm** [comment]
I'm not the right person to ask about cross issues. Maybe @amjoseph-nixpkgs or @samueldr ?

**Comment by Atemu** [comment]
Note that you're cross-compiling to an exotic [Tier 4 platfrom](https://github.com/NixOS/rfcs/blob/master/rfcs/0046-platform-support-tiers.md#tier-4) here. It is not generally supported. Do not ping general package maintainers about its issues.

**Comment by axelkar** [comment]
> Note that you're cross-compiling to an exotic [Tier 4 platfrom](https://github.com/NixOS/rfcs/blob/master/rfcs/0046-platform-support-tiers.md#tier-4) here. It is not generally supported. Do not ping general package maintainers about its issues.

Sorry! I had no idea! Thanks for telling me.

**Comment by amjoseph-nixpkgs** [comment]
Yeah, I mean, `aarch64-android` is like quadruply-exotic-extraterrestrial voodoo for nixpkgs.

I'm afraid I can't really help with this.  I've never done any android stuff with nixpkgs at all.
